src/main/java/com/google/devtools/build/lib/runtime/BlazeCommandDispatcher.java

```diff
@@ -781,14 +775,23 @@ private BlazeCommandResult execExclusively(
   }
 
   private static class RemoteCacheTransientErrorException extends IOException {
+    // Remains constant across retries.
+    private final String buildId;
```

Comment:
nit: please rename this field to `buildRequestId` to make it consistent with the field in `CommandEnvironment`. Same for `buildIdOverride`.